src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -451,8 +441,6 @@ otError Leader::RlocLookup(uint16_t  aRloc16,
             subCur = prefix->GetSubTlvs();
             subEnd = prefix->GetNext();
 
-            VerifyOrExit(subEnd <= end, error = OT_ERROR_PARSE);
```

Comment:
Adding a note about this:

This check is not needed (is redundant) since it is covered by 
```
   VerifyOrExit((cur + 1) <= end && cur->GetNext() <= end, error = OT_ERROR_PARSE);
``` 
few lines above. Note that `subEnd = prefix->GetNext()` and `prefix` is same as `cur`.